imap_processing/ultra/l1b/ultra_l1b_extended.py

```diff
@@ -111,6 +123,115 @@ def get_front_y_position(start_type: ndarray, yb: ndarray) -> tuple[ndarray, nda
     return np.array(d), np.array(yf)
 
 
+def get_ph_tof_and_back_positions(
+    de_dataset: xarray.Dataset, xf: np.ndarray, sensor: str
+) -> tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
```

Comment:
No need to do this, but thought I'd throw this suggestion out in case you like it.

Would it be useful to stick to datasets since these are all the same size and in a sense grouped together? Then you could return a single object that contains all of these arrays.

```suggestion
def get_ph_tof_and_back_positions(
    de_dataset: xarray.Dataset, xf: np.ndarray, sensor: str
) -> xarray.Dataset:
```